cli/src/cli_util.rs

```diff
@@ -1455,15 +1432,28 @@ See https://github.com/martinvonz/jj/blob/main/docs/working-copy.md#stale-workin
             .get_wc_commit_id(self.workspace_id())
             .map(|commit_id| tx.base_repo().store().get_commit(commit_id))
             .transpose()?;
+        let maybe_new_wc_commit = tx
+            .repo()
+            .view()
+            .get_wc_commit_id(self.workspace_id())
+            .map(|commit_id| tx.repo().store().get_commit(commit_id))
+            .transpose()?;
         if self.working_copy_shared_with_git {
             let git_repo = self.git_backend().unwrap().open_git_repo()?;
-            self.export_head_to_git(tx.mut_repo(), &git_repo)?;
+            if let Some(wc_commit) = &maybe_new_wc_commit {
+                git::reset_head(tx.mut_repo(), &git_repo, wc_commit)?;
+            }
             let failed_branches = git::export_refs(tx.mut_repo(), &git_repo)?;
             print_failed_git_export(ui, &failed_branches)?;
         }
         self.user_repo = ReadonlyUserRepo::new(tx.commit());
         if self.may_update_working_copy {
```

Comment:
unrelated to this PR: it looks like we can delete this check, given the assertion at the beginning of the function

Reply:
Do you mean moving the check to `update_working_copy()`?

Reply:
Indeed. I was tricked by the skipped context.